README.md

```diff
@@ -1,2 +1,344 @@
+require LocalStrategy = require('passport-local').Strategy
+```
+
+We've already imported the local strategy middlware module, so all we really need to do at this point is open up `index.js` and somewhere below where our models are defined, define the specific validation logic that suits our application. This can be done like so:
+
+```
+passport.use(new LocalStrategy((username, password, done) => {
+  User
+    .forge({ username: username })
+    .fetch()
+    .then((usr) => {
+      if (!usr) {
+        return done(null, false);
+      }
+      usr.validatePassword(password).then((valid) => {
+        if (!valid) {
+          return done(null, false);
+        }
+        return done(null, usr);
+      });
+    })
+    .catch((err) => {
+      return done(err);
+    });
+}));
+```
+
+So now we've defined our local strategy. You may very well be wondering how all this fits together, and we'll get to that. But first let's examine what we are doing in our strategy.
+
+Our first step, as you can see, is to create a new Strategy object (`new LocalStrategy`) and then register that with passport via `passport.use()`. However, importantly, when we create the new strategy we pass into the constructor a callback, let's call it our validation function, that defines the *specific way that our application will handle validation.*
+
+This validation function is hugely important to how Passport works, as well as why Passport is such a good library.  Remember above, when we talked about how Passport is unopninated about how an application performs its validation. Well, in addition to being able to choose which strategies an application uses to validate a user, Passport is also unopinionated about what happens during validation within a given strategy! It simply expects a strategy to be defined and for the application itself to provide the validation logic. Pretty clever. 
+
+So what is our validation logic? Well, it's pretty straightforward it turns out. The callback we've supplied above takes the username and password, which the user will have supplied when they attempt to login, and then a callback that we've called done.  This callback is important because it is what we'll use to hand control to the next step in server's handling of the request. As we shall see, it has the following signature: `done(error, user[, msg])`.
+
+Inside the function we use our Bookshelf User model to try to fetch a user using the supplied username. Then, if the user is not found we call `done(null, false)` to indicate that there was no error (so `null` for the error argument), but no user was found (so `false` for the user argument). If a user was found, then we try to validate the password using the password validation function that we added to our User model. If the passowrd is invalid, we again call `done(null, false)`. Otherwise, we call `done(null, usr)`passing the usr that we've found to the next step in the process. Finally, if an error occurs we simply call `done(err)`.
```

Comment:
I like this -- could you add an explanation about why students shouldn't error out on an incorrect password but should instead (as you've done in this example) just return no user?